src/readme.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 Microsoft Corporation
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+import * as commonmark from 'commonmark'
+import { parse } from '@ts-common/commonmark-to-markdown'
+import { getTagsToSettingsMapping, getCodeBlocksAndHeadings } from '@azure/openapi-markdown'
+import * as fs from 'fs'
+import { safeLoad } from './utils'
+
+/**
+ * walks a markdown tree until the callback provided returns true for a node
+ */
+export const walkToNode = (
+  walker: commonmark.NodeWalker,
+  cb: (node: commonmark.Node) => boolean,
+): commonmark.Node | undefined => {
+  let event = walker.next()
+
+  while (event) {
+    const curNode = event.node
+    if (cb(curNode)) {
+      return curNode
+    }
+    event = walker.next()
+  }
+  return undefined
+}
+
+export const sortByApiVersion = (versions: string[]): string[] => {
+  // sort by api version. Format: YYYY-MM-DD
+  const supportedRegex = [/(\d{4})-(\d{2})-(\d{2})/, /(\d{4})_(\d{2})_(\d{2})/]
```

Comment:
if we want to support `YYYY_MM_DD`, how about just using `version.replaceAll('_', '-')`?

Reply:
We just define current supported format here. And for YYYY_MM_DD. There will be another avocado rule to report it. It should not be a valid case.